cmd/lakectl/cmd/root_test.go

```diff
@@ -66,3 +69,92 @@ func TestLakectlUnixPerm(t *testing.T) {
 		})
 	}
 }
+
+func TestInitConfig_LoadingScenarios(t *testing.T) {
+	type setupFn func(t *testing.T) (cleanup func())
+
+	tests := []struct {
+		name           string
+		setup          setupFn
+		expectedSource string
+		expectedURL    string
+	}{
+		{
+			name: "using --config flag",
+			setup: func(t *testing.T) func() {
+				tmpFile := filepath.Join(t.TempDir(), "config.yaml")
+				content := "server:\n  endpoint_url: \"http://flag-endpoint\"\n"
+				require.NoError(t, os.WriteFile(tmpFile, []byte(content), 0644))
+				cfgFile = tmpFile
+				return func() {
+					cfgFile = ""
+				}
+			},
+			expectedSource: "http://flag-endpoint",
+			expectedURL:    "http://flag-endpoint",
+		},
+		{
+			name: "using LAKECTL_CONFIG_FILE env var",
+			setup: func(t *testing.T) func() {
+				tmpFile := filepath.Join(t.TempDir(), "env_config.yaml")
+				content := "server:\n  endpoint_url: \"http://env-endpoint\"\n"
+				require.NoError(t, os.WriteFile(tmpFile, []byte(content), 0644))
+				t.Setenv("LAKECTL_CONFIG_FILE", tmpFile)
+				cfgFile = ""
+				return func() {
+					os.Unsetenv("LAKECTL_CONFIG_FILE")
+				}
+			},
+			expectedSource: "http://env-endpoint",
+			expectedURL:    "http://env-endpoint",
+		},
+		{
+			name: "using home directory default",
+			setup: func(t *testing.T) func() {
+				homeDir := t.TempDir()
+				tmpFile := filepath.Join(homeDir, ".lakectl.yaml")
+				content := "server:\n  endpoint_url: \"http://home-endpoint\"\n"
+				require.NoError(t, os.WriteFile(tmpFile, []byte(content), 0644))
+				t.Setenv("HOME", homeDir)
+				cfgFile = ""
+				return func() {
+					os.Unsetenv("HOME")
+				}
+			},
+			expectedSource: "http://home-endpoint",
+			expectedURL:    "http://home-endpoint",
+		},
+		{
+			name: "env-only fallback",
+			setup: func(t *testing.T) func() {
+				t.Setenv("LAKECTL_SERVER_ENDPOINT_URL", "http://env-only")
+				cfgFile = ""
+				return func() {
+					os.Unsetenv("LAKECTL_SERVER_ENDPOINT_URL")
+				}
+			},
+			expectedSource: "",
+			expectedURL:    "http://env-only",
+		},
```

Comment:
This test func should only test lakectl configuration file loading, no need to test the env var fallback mechanism